### PR TITLE
improve swapd state management 

### PR DESCRIPTION
**Pull request by zkao**

### CODE REVIEW

File: src/swapd/runtime.rs [hunk 1]
```diff
@@ -838,39 +907,27 @@ impl Runtime {
...
-
-                        match self.state.swap_role() {
-                            SwapRole::Alice => {}
-                            SwapRole::Bob => {
-                                let addr = self
-                                    .state
-                                    .b_address()
-                                    .cloned()
-                                    .expect("address available at CommitB");
-                                let msg = format!(
```
Comment (TheCharlatan) [inline src/swapd/runtime.rs:857]: This message is no longer pushed to the progress report. Is your goal to put the funding address and amount under a different cli endpoint? Like it stands currently I cannot drive the swap with just the cli.

Comment (zkao) [reply src/swapd/runtime.rs:857]: ops, i guess i forgot to push

Comment (zkao) [reply src/swapd/runtime.rs:857]: @TheCharlatan can you report precisely the exact conditions you don't get the stuff pushed to the progress report?

Comment (zkao) [reply src/swapd/runtime.rs:857]: 6d26a61 is the correct commit i tested

Comment (TheCharlatan) [reply src/swapd/runtime.rs:857]: No, you are right. Progress shows up as before. Maybe I forgot to pull, sorry!

**Review by TheCharlatan — APPROVED** [review]
tACK ce36f882e983adca541a42a1aee3fd266de04eb3

**Review by Lederstrumpf — APPROVED** [review]
tACK - thanks for getting all that convoluted `remote_commit` matching out!

**Pull request merged**